core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```
@@ -383,10 +387,13 @@ object ConsoleConsumer extends Logging {
       KafkaMetricsReporter.startReporters(verifiableProps)
     }
 
-    //Provide the consumer with a randomly assigned group id
-    if (!consumerProps.containsKey(ConsumerConfig.GROUP_ID_CONFIG)) {
-      consumerProps.put(ConsumerConfig.GROUP_ID_CONFIG, s"console-consumer-${new Random().nextInt(100000)}")
+    if (options.has(groupIdOpt))
+      // group id provided as an argument takes precedence over one provided in the config file
+      extraConsumerProps.put(ConsumerConfig.GROUP_ID_CONFIG, options.valueOf(groupIdOpt))
```

What if the user had provided a group via `extraConsumerProps`?

---

This line runs only if there is a `--group` argument provided in the command line. Command line options take precedence over those provided in the config file. Therefore, it's safe to overwrite the possible value provided in the config file if the same option is also provided as a command line argument.

---

Rather than enforcing precedence, we could throw an exception if there is an inconsistency. Seems weird for the user to try and specify groupId in multiple places.

---

@hachikuji Thanks for the review. I submitted an update to allow providing `groupId` in one place only.